amazonia/classes/asg.py

```diff
@@ -106,16 +106,18 @@ def create_asg(self, title, minsize, maxsize, subnets, load_balancer, keypair, i
                 self.sns_notification_configurations = self.trop_asg.NotificationConfigurations = \
                     [NotificationConfigurations(TopicARN=sns_topic_arn, NotificationTypes=sns_notification_types)]
             else:
-                raise MalformedSNSError("Error: sns_notification_types must be a non null list.")
+                raise MalformedSNSError('Error: sns_notification_types must be a non null list.')
 
         self.trop_asg.LaunchConfigurationName = Ref(self.create_launch_config(
             title=title,
             keypair=keypair,
             image_id=image_id,
             instance_type=instance_type,
             iam_instance_profile_arn=iam_instance_profile_arn,
-            userdata=userdata if userdata is not None else ""
```

Comment:
I thought it might be better to remove the userdata=userdata line (117) here entirely and then just assign it in the if statement below?

eg: `self.trop_asg.LaunchConfigurationName.userdata = userdata if userdata is not None`